cmd/vulcan-gitleaks/gitleaks.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2019 Adevinta
+*/
+
+package main
+
+import (
+	"context"
+
+	"github.com/adevinta/vulcan-check-sdk/helpers/command"
+	"github.com/sirupsen/logrus"
+)
+
+const Cmd = `gitleaks`
+
+var params = []string{"detect", "/tmp/repo", "-f", "json", "-r", reportOutputFile, "--no-git"}
```

Comment:
One suggestion here, but please feel free to discard it. 

Instead of running the tool with the default and embedded rule set, why not passing a custom [.gitleaks.toml](https://github.com/zricethezav/gitleaks/blob/master/config/gitleaks.toml) file? The advantages I see are:
1. we can easily disable/customise wanky rules (and that's actually the case for [this one rule](https://github.com/zricethezav/gitleaks/blob/master/config/gitleaks.toml#L627-L640) that I personally keep disabled due to the sheer amount of false positive it raises. We can talk about this specific rule in a different thread if you want)
2. more importantly, we can easily add our own custom rules to the stack. I actually have a working example I've had to use for the pentest of Fotocasa.pro:

   ```
    [[rules]]
    id = "sql-connection-string"
    description = "Connection string for a SQL database"
    regex = '''(?m)connectionString=".*password=([^;]+)'''
    secretGroup = 1
   ```
    (yes, I know - I should submit a PR upstream for this, but I am lazy 😅)

The only disadvantage I see for this approach is that getting the latest updates from the upstream version of `.gitleaks.toml` file becomes then a manual job.

@everyone - thoughts?

Reply:
I don't have a strong opinion on this and I am personally okay with using the embedded rule set. However, we have an example of a check that functions exactly how you describe: https://github.com/adevinta/vulcan-checks/blob/master/cmd/vulcan-exposed-http-resources/main.go#L135-L145
In any case, I think accepting rules to be supplied through the options can always be useful.

Reply:
Another option would be to add a custom `.gitleaks.toml` file to the workdir of the Docker image, and the tool will automatically use this file as the ruleset. This would require us to review changes in the upstream configuration file to update our custom file with latest rules.
Allowing additional rules to be provided through options can be tricky but it's possible, although I don't see a clear advantage of doing so in front of creating a custom ruleset file, taking into account the complexity this can add to the check manteinance.
Regarding the rule disabling, it is currently supported through the ExcludedRules option.

Reply:
Related to [Julian's comment](https://github.com/adevinta/vulcan-checks/pull/185#discussion_r847348150), I think it's fine to use the predefined rule list for now (I mean, this is purely a product decision), this way we also avoid the toil of manually merging changes from upstream (which, as Julian mentioned, it doesn't really work for us). 

This commend can be resolved I think.